### ISSUE TITLE
Boosting rounds missing from XGBClassifier docs

### ISSUE THREAD

**Issue by david-cortes** [dmlc/xgboost#6012]
The docs for `XGBClassifier` (https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier) do not have any parameter corresponding to the number of boosting rounds.

As far as I am aware, the number of boosting rounds is still controllable when using `binary:logistic` loss, and if I'm not mistaken, that XGBClassifier class should just be a wrapper around XGBModel (https://github.com/dmlc/xgboost/blob/674c409e9d24b496d925781a3345b626efbf9dd9/python-package/xgboost/sklearn.py#L201), which does support an `n_estimators` parameter.

**Comment by trivialfis** [comment]
There is no num boost round for skl interface, only n estimator.  It's equivalent, just different jargons that we are not able to change.

**Comment by david-cortes** [comment]
@trivialfis : thing is, there is no mention of `n_estimators` in the docs either.
